### PR TITLE
ROC plotter works with any number of trained models

### PR DESCRIPTION
**Pull request by Aylr**
## Notes

Note there are some known issues (coloring is broken, legend is missing) that are represented in #237 that will be fixed later in effort to keep PRs reasonable and prioritize work.

## Things to test

- [x] Run example_simple_classification.py and see a single ROC plot from a TSM (trained super...)
- [x] Try to run it from example_simple_regression.py and see a helpful error
- [x] Also note that TSM now has a few more handy properties

### CODE REVIEW

File: healthcareai/develop_supervised_model.py [hunk 1]
```diff
@@ -491,6 +491,12 @@ def random_forest(self, cores=4, trees=200, tune=False, debug=False):
     def _trainer(self, algorithm, include_factor_model=True):
         # TODO should the factor model be either 1) optional or 2) separate?
         algorithm.fit(self.X_train, self.y_train)
+
+        if self.model_type == 'classification':
+            self.test_set_predicted = algorithm.predict_proba(self.X_test)
+        elif self.model_type == 'regression':
+            self.test_set_predicted = algorithm.predict(self.X_test)
+
```
Comment (mmastand) [inline healthcareai/develop_supervised_model.py]: Nice.

File: healthcareai/trained_models/trained_supervised_model.py [hunk 2]
```diff
@@ -24,20 +26,35 @@ def __init__(self,
...
         self.column_names = column_names
-        self.model_type = model_type
+        self._model_type = model_type
         self.grain_column = grain_column
         self.prediction_column = prediction_column
-        self.y_pred = y_pred
-        self.y_actual = y_actual
+        self.test_set_predictions = test_set_predictions
+        self.test_set_actual = test_set_actual
         self._metric_by_name = metric_by_name
```
Comment (mmastand) [inline healthcareai/trained_models/trained_supervised_model.py]: I like that these live in the test set as well. Is it possible to get all the predictions back in the development step?

File: example_simple_classification.py [hunk 3]
```diff
@@ -108,14 +109,18 @@ def main():
...
     # Look at the RF feature importance rankings
     # TODO this is broken
     # hcai.get_advanced_features().plot_rffeature_importance(save=False)
 
     # Create ROC plot to compare the two models
     # TODO this is broken - it might look like tools.plot_roc(models=[random_forest, linear, knn])
-    # hcai.plot_roc()
+
+    # Create a single ROC plot from the trained model
+    trained_model.roc_curve_plot()
```
Comment (mmastand) [inline example_simple_classification.py]: If you pass `trained_ensemble` into this, it shows you just the best model. Would we want to serve up all of the curves for compare?

File: example_simple_classification.py [hunk 4]
```diff
@@ -108,14 +109,18 @@ def main():
...
+
+    # Create a comparison ROC plot multiple models
+    hcaieval.tsm_comparison_roc_plot([trained_random_forest, trained_knn, trained_logistic_regression])
```
Comment (mmastand) [inline example_simple_classification.py]: Needs a legend and separate colors.

File: example_simple_classification.py (hunk 3 above)
Comment (Aylr) [inline example_simple_classification.py]: Oh, that's a great idea!
I'll have to think about where to implement that. Noted in #239

File: example_simple_classification.py (hunk 4 above)
Comment (Aylr) [inline example_simple_classification.py]: Please see #237 :smile:

**Pull request merged**